x/wasm/ioutils/utils.go

```diff
@@ -17,7 +17,7 @@ var (
 
 // IsGzip returns checks if the file contents are gzip compressed
 func IsGzip(input []byte) bool {
-	return bytes.Equal(input[:3], gzipIdent)
+	return len(input) > 3 && bytes.Equal(gzipIdent, input[0:3])
```

Comment:
```suggestion
	return len(input) >= 3 && bytes.Equal(gzipIdent, input[0:3])
```

Before we accepted a length of 3 as valid gzip. I'd keep it that way and leave it up to the uncompressor to check everything else. Otherwise we should add a comment how much data we require after the `gzipIdent` and why.